docs/controller.md

```diff
@@ -0,0 +1,141 @@
+>[!Note]
+> We chose a `configuration` section in our manifest because maybe in the future we will add a `variable` section to specify environment variables.
+#### `actions.<action_id>.configuration.container`
+Here, you will declare the address of the container you want to run your action on.
+>[!Note]
+> Maybe in the future we are going to provide virtual machine support to run your actions, that is the reason why we named this part `container` and not for instance `os`.
+
+**Example :**
+```yaml
+actions:
+  postinstall:
+    configuration:
+      container: debian:latest
+```
+#### `actions.<action_id>.commands`
+`command` is a **list** of shell commands that will be executed during the action.
+**Example**
+```yaml
+actions:
+  postinstall:
+    configuration:
+      container: debian:latest
+    commands:
+      - apt update
+      - apt install mfa-postinstall
+```
+### HTTP Request (Input)
+A controller triggers a pipeline once it receives its corresponding manifest. To do so, an HTTP client needs to send a POST request containing the manifest file and the name of the pipeline.
+
+- `POST` /pipeline :
+
+  **Body**:
+    - `name` : a `string` that corresponds to the pipeline name.
+
+    - `body` : a `file` that is the manifest file conform to the structure declared bellow.
+
+>[!Note]
+> The request **must** be a multipart/form-data since the pipeline file could be quite long.
+
+### HTTP Response (Output)
+The pipeline needs to inform the user on the state of the actions, therefore it needs to provide outputs. Outputs aim to describe each actions state to get an insight on what is going on in your pipeline. An output has an **header** that must have one of the following value :  `PENDING`, `SCHEDULED`, `DOING`, `COMPLETED` and `ERROR`.
+#### States
+- `PENDING` : the action has not been sent to the Scheduler yet.
+
+  **Payload** : none.
+
+- `SCHEDULED`: the action has been received by the Scheduler but has not been assigned to an Agent.
+
+  **Payload** : none.
+
+- `RUNNING` : the action has been assigned to an Agent but not completed.
+
+  **Payload** : logs from the agent (these logs can change during the execution of the action so they need to be re-fetched to be up to date).
+
+- `COMPLETED` : the action has finished successfully.
+
+  **Payload** : none.
+
+- `ERROR`: the action has completed but encountered an error.
+
+  **Payload** : a message detailing the error.
+
+## Diagrams
+### Sequence diagram
+
+```mermaid
+sequenceDiagram
+    actor User
+    participant HTTPClient
+    participant Controller
+    participant Scheduler
+
+    participant Database
+
+    HTTPClient->>Controller: URL + Action
+    Controller->>HTTPClient: Acknowledgment
+
+    alt is request malformed
+        Controller-->>User: Nok
+    else is well
+        Controller-->>User: Ok
+    end
+
+    Controller->>Database: saves pipeline in database
+
+    loop over action steps
```

Comment:
A 'step' is not defined vocabulary. It is simply command to execute.

(an 'action' is a list of commands + environment)

Reply:
Resolved